### ISSUE TITLE
Data mining to fetch latest pal + trait stats

### ISSUE THREAD

**Issue by tylercamp** [tylercamp/palcalc#6]
Pal Calc depends on a list of pal data from the game in order to do its breeding calculations. It also uses some trait data for displaying traits in its UI.

Pal Calc currently uses data mined by others and shared publicly. Unfortunately there isn't a consistent source of information which has _all_ of the data needed for its calculations.

How is this data mining performed, and can it be done as part of Pal Calc?

Alternatively, is there another frequently-updated source that could be referenced? We will need to collect, at least, the data mentioned in [PalCal.GenDB](https://github.com/tylercamp/palcalc/blob/main/PalCalc.GenDB/).

https://paldb.cc was [suggested](https://www.reddit.com/r/Palworld/comments/1ccwnap/comment/l18bhwk/) and is frequently updated with the list of pals, but it seems to be missing `InternalIndex` which is required as a "tie-breaker" for breeding calculations.

**Comment by RudelNudel** [comment]
Hi,

under https://paldb.cc/en/Breeding_Farm the InternalIndex for Pals is shown as IndexOrder.

**Comment by tylercamp** [comment]
Thanks for the reference @RudelNudel ! I didn't see this before

paldb.cc tends to be updated regularly so I'm ok with relying on that as a source. The only remaining question is a DB for traits/passives. paldb.cc includes a list of traits and their ranks, but we'll also need the internal "code-name" for each trait in order to recognize them in save files

**Comment by tylercamp** [comment]
This was mentioned as a possible source for palworld data: https://github.com/KrisCris/Palworld-Pal-Editor/blob/develop/src/palworld_pal_editor/assets/data/

It just seems to be missing the `InternalIndex`.

I also took another stab at data-mining using `FModel` based on the instructions here: https://pwmodding.wiki/docs/intro

I was able to extract the latest set of pal icons and the mapping of internal->display names for skills, but I wasn't able to properly extract the list of skills or the list of pals. I exported the `.uasset` files and could see the info I wanted in a text editor, but it seems to be a UE-specific table format.

---

In the meantime, paldb.cc for full pal info and `Palworld-Pal-Editor` for trait IDs seems to be a sufficient combo for now. I'm manually updating with new pals + traits and will write an exporter to scrape these later.
